### PR TITLE
test(anvil): add eth_getLogs test

### LINKED ISSUE #1515: Anvil `eth_getLogs` returns no logs
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (eb4bfb2 2022-05-05T18:06:46.302249388+00:00)

### What command(s) is the bug in?

anvil

### Operating System

Linux

### Describe the bug

The [`get_past_events`](https://github.com/gakonst/ethers-rs/blob/2768a6b4d7dee812d7de0a1e336214bd46e9bacb/ethers-contract/tests/contract.rs#L98) test in `ethers-rs` fails when using Anvil because no logs are returned by `eth_getLogs`. Anvil is run with no parameters and no config, so all the defaults are used.

The transaction does result in a receipt and a block, and the blog does have logs:

**`eth_getLogs` call trace**

```
2022-05-05T18:27:17.231067Z TRACE rpc: handling call RpcMethodCall { jsonrpc: V2, method: "eth_getLogs", params: Array([Object({"address": String("0x5fbdb2315678afecb367f032d93f642f64180aa3"), "fromBlock": String("0x0"), "topics": Array([String("0x999b6d464c4e3383c341bdd3a22b02dda8a7e1d69c069d252e35cb2ee2f4a3c3"), String("0x00000000000000000000000015d34aaf54267db7d7c367839aaf71a00a2c6a65")])})]), id: Number(12) }
2022-05-05T18:27:17.231077Z TRACE rpc: received method call RpcMethodCall { jsonrpc: V2, method: "eth_getLogs", params: Array([Object({"address": String("0x5fbdb2315678afecb367f032d93f642f64180aa3"), "fromBlock": String("0x0"), "topics": Array([String("0x999b6d464c4e3383c341bdd3a22b02dda8a7e1d69c069d252e35cb2ee2f4a3c3"), String("0x00000000000000000000000015d34aaf54267db7d7c367839aaf71a00a2c6a65")])})]), id: Number(12) }
2022-05-05T18:27:17.231091Z TRACE rpc: received handler request EthGetLogs(Filter { from_block: Some(Number(0)), to_block: None, block_hash: None, address: Some(Value(0x5fbdb2315678afecb367f032d93f642f64180aa3)), topics: Some(Value(Some(Array([Some(0x999b6d464c4e3383c341bdd3a22b02dda8a7e1d69c069d252e35cb2ee2f4a3c3), Some(0x00000000000000000000000015d34aaf54267db7d7c367839aaf71a00a2c6a65)])))) })
2022-05-05T18:27:17.231099Z TRACE rpc::api: executing eth request EthGetLogs(Filter { from_block: Some(Number(0)), to_block: None, block_hash: None, address: Some(Value(0x5fbdb2315678afecb367f032d93f642f64180aa3)), topics: Some(Value(Some(Array([Some(0x999b6d464c4e3383c341bdd3a22b02dda8a7e1d69c069d252e35cb2ee2f4a3c3), Some(0x00000000000000000000000015d34aaf54267db7d7c367839aaf71a00a2c6a65)])))) })
2022-05-05T18:27:17.231155Z TRACE rpc: prepared rpc result Success(Array([]))
```

**Receipt for transaction that generates the logs**

```
2022-05-05T18:27:17.230396Z TRACE rpc: prepared rpc result Success(Object({"blockHash": String("0xadfa16650cdc51587c00e887f273fd43d89d3cdd5a898af24227148d25b861ff"), "blockNumber": String("0x2"), "contractAddress": Null, "cumulativeGasUsed": String("0x0"), "effectiveGasPrice": String("0x4a817c800"), "from": String("0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"), "gasUsed": String("0xdcf2"), "logs": Array([Object({"address": String("0x5fbdb2315678afecb367f032d93f642f64180aa3"), "blockHash": String("0xadfa16650cdc51587c00e887f273fd43d89d3cdd5a898af24227148d25b861ff"), "blockNumber": String("0x2"), "data": String("0x00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000d696e697469616c2076616c75650000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000026869000000000000000000000000000000000000000000000000000000000000"), "logIndex": String("0x0"), "removed": Bool(false), "topics": Array([String("0x999b6d464c4e3383c341bdd3a22b02dda8a7e1d69c069d252e35cb2ee2f4a3c3"), String("0x000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb92266"), String("0x0000000000000000000000000000000000000000000000000000000000000000")]), "transactionHash": String("0xce0152e34e73c7b054219e6b3f15a54f629f606b4eba9ad747b45e225efc597a"), "transactionIndex": String("0x0"), "transactionLogIndex": String("0x0")})]), "logsBloom": String("0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040020000000000000100000800080000008000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000040000000200000000000000000000000002000000000000000000021000000000000000000000000000000000000000000000000000000000000000000"), "root": Null, "status": String("0x1"), "to": String("0x5fbdb2315678afecb367f032d93f642f64180aa3"), "transactionHash": String("0xce0152e34e73c7b054219e6b3f15a54f629f606b4eba9ad747b45e225efc597a"), "transactionIndex": String("0x0")}))
```

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Closes #1515

## Solution
* replicate get logs test successfully

@onbjerg replicated the test successfully, perhaps the issue of #1515 is with the address topic somehow

### CODE REVIEW

File: anvil/tests/it/transaction.rs [hunk 1]
```diff
@@ -304,3 +315,40 @@ contract Contract {
...
+
+    // 2 events, 1 in constructor, 1 in call
+    assert_eq!(logs[0].new_value, "initial value");
+    assert_eq!(logs[1].new_value, "hi");
+    assert_eq!(logs.len(), 2);
+
+    // and we can fetch the events at a block hash
+    let hash = client.get_block(1).await.unwrap().unwrap().hash.unwrap();
+
+    let logs: Vec<ValueChanged> =
```
Comment (0xvv) [inline anvil/tests/it/transaction.rs:350]: Nitpicking, but could be named `logs_at_block_1` to make the assertions at line 352 and 353 clearer and to not have two vars with the same name

**Review by 0xvv — APPROVED** [review]
Looks good, just a smöl suggestion

**Comment by onbjerg** [comment]
That's strange.. The test passed on Ganache. Maybe Ganache has a bug then and it just somehow worked?

**Review by onbjerg — APPROVED** [review]

**Comment by mattsse** [comment]
not sure but maybe chain id related?

**Pull request merged**